### PR TITLE
Add GHA to sync pytorch-canary

### PR DESCRIPTION
**Pull request by janeyx99**
This would run every 15 minutes to sync pytorch-canary with pytorch master.

### CODE REVIEW

File: .github/workflows/sync_pytorch_canary.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: sync_pytorch_canary
+
+on:
+  schedule:
+    # Every day at 3:33am ET
+    - cron: "33 7 * * *"
```
Comment (seemethere) [inline .github/workflows/sync_pytorch_canary.yml:6]: nit: Should we have it run more frequently?

**Review by seemethere — APPROVED** [review]

File: .github/workflows/sync_pytorch_canary.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      - name: Set up SSH
+        env:
+          PYTORCH_CANARY_SSH_PRIVATE_KEY: ${{ secrets.PYTORCH_CANARY_SSH_PRIVATE_KEY }}
+        run: |
+          set -eu
+          mkdir -p ~/.ssh
+          echo "$PYTORCH_CANARY_SSH_PRIVATE_KEY" > ~/.ssh/id_rsa
+          chmod 600 ~/.ssh/id_rsa
+      - name: Clone pytorch/pytorch
+        run: git clone --bare https://github.com/pytorch/pytorch.git
```
Comment (janeyx99) [inline .github/workflows/sync_pytorch_canary.yml:24]:
```suggestion
        run: git clone --bare https://github.com/pytorch/pytorch.git pytorch
```

File: .github/workflows/sync_pytorch_canary.yml (hunk 1 above)
Comment (driazati) [reply .github/workflows/sync_pytorch_canary.yml:6]: agree, the workflow should be pretty light so even something like every 15 minutes seems like it'd be fine

File: .github/workflows/sync_pytorch_canary.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+name: sync_pytorch_canary
+
+on:
+  schedule:
+    # Every 15 minutes
+    - cron: "*/15 * * * *"
```
Comment (janeyx99) [inline .github/workflows/sync_pytorch_canary.yml:6]: I wanted to do something more unique, like "03/15 * * * *", but that's nonstandard and I don't know if GHA will handle it properly.

**Pull request merged**